### ISSUE TITLE
Working with more than one Coq file does not work

### ISSUE THREAD

**Issue by MSoegtropIMC** [coq-community/vscoq#155]
Using VSCoq with just one Coq file open works reasonably well.
When I open a second Coq file it behaves fairly erratically. Some observable behavior:
- open file1.v + step down 1 step
- open file2.v + step down 1 step
- close file2.v
  * file1.v is gone (closed)
  * reopen file1.v shows that it has its state lost
  * step down in file1.v reopens file2.v
  * closing file2.v and reopen file1.v and step 1 step down again opens file2.v
  * As far as I can tell the only way out of this is to quit and reopen VSCode
- closing all open files before opening another one also doesn't help to avoid this.

In summary: I see now way to work with more than one Coq file except closing and reopening VSCode for each file.

**Comment by fakusb** [comment]
For the record: There is a  difference in clicking and double-clicking a file in the explorer bar left: A single-clicked file is closed when focus is lost (and the file name is in italic), a double clicked file is left open, maybe that was part of the problem.

Furthermore, `F1` and then `Reload Windows` is my workaround when a file got closed by me by accident. It is a bit faster than closing and reopening vscode.

**Comment by TheoWinterhalter** [comment]
When a file is in italic, you can also double click on the tab to make it persistent or just simply save.

**Comment by fakusb** [comment]
Why can't we simply comment out those two lines that throw the exception that prevents re-opening a file to fix the problem?
https://github.com/coq-community/vscoq/blob/f587c15aaa3d3adef3c2e87f429328b2d1906942/client/src/CoqLanguageServer.ts#L131

One would loose the progress in the file, but that is to be expected if one closes the file.

Is it because if done so, the language server in the background still has some Resources and that would leak memory?
So, would a proper fix need to "remove" the CoqDocument from the language server when a file is closed?

**Comment by fakusb** [comment]
(I moved the discussion on my proposed "fix" to #167  )
